React/Base/RCTConvert.m

```diff
@@ -913,8 +913,8 @@ + (UIImage *)UIImage:(id)json
 
   if (!CGSizeEqualToSize(imageSource.size, CGSizeZero) &&
       !CGSizeEqualToSize(imageSource.size, image.size)) {
-    RCTLogError(@"Image source size %@ does not match loaded image size %@.",
-                NSStringFromCGSize(imageSource.size), NSStringFromCGSize(image.size));
+    RCTLogError(@"Image source %@ expects size %@ does not match loaded image size %@.",
+                imageSource.imageURL, NSStringFromCGSize(imageSource.size), NSStringFromCGSize(image.size));
```

Comment:
Good idea, but might be better just to use `imageURL.path.lastPathComponent` here, since the prefix is mostly noise.